src/wallets/hardware/bcvault/index.js

```diff
@@ -0,0 +1,155 @@
+import { Transaction } from 'ethereumjs-tx';
+import { bufferToHex } from 'ethereumjs-util';
+import * as bc from 'bc-vault-js';
+import { BCVAULT as bcVault } from '../../bip44/walletTypes';
+import HDWalletInterface from '@/wallets/HDWalletInterface';
+import errorHandler from './errorHandler';
+import BigNumber from 'bignumber.js';
+import commonGenerator from '@/helpers/commonGenerator';
+import store from '@/store';
+
+import {
+  getBufferFromHex,
+  sanitizeHex,
+  getSignTransactionObject,
+  calculateChainIdFromV
+} from '../../utils';
+
+const NEED_PASSWORD = false;
+
+class BCVault {
+  constructor() {
+    this.identifier = bcVault;
+    this.isHardware = true;
+    this.needPassword = NEED_PASSWORD;
+    this.selectedAddress = '';
+    this.bcWallet = new bc.BCJS(() => {});
+    this.deviceNumber = null;
+    this.bcWalletType = bc.WalletType.ethereum;
+  }
+
+  async init() {
+    // fetch devices
+    this.deviceNumber = await this.bcWallet.getDevices().catch(errorHandler);
+    // get wallet of first device and password
+    // not sure if we want the users to pass this as a parameter or ask user
+    // for which wallet to use
+    await this.bcWallet
+      .EnterGlobalPin(this.deviceNumber[0], this.bcWalletType)
+      .catch(errorHandler);
+    const walletAddresses = await this.bcWallet.getBatchWalletDetails(
+      this.deviceNumber[0],
+      [bc.WalletType.ethereum]
+    );
+
+    return walletAddresses;
+  }
+
+  getAccount(address) {
+    this.selectedAddress = address;
+    const path = null;
+    const publickey = address;
+    const txSigner = async tx => {
+      if (store.state.main.network.type.chainID === 1) {
+        delete tx['from'];
+        tx['from'] = address;
+        tx = new Transaction(tx, {
+          common: commonGenerator(store.state.main.network)
+        });
+        const newTx = {};
+        newTx['feeCount'] = new BigNumber(
+          bufferToHex(tx['gasLimit'])
+        ).toNumber();
+        newTx['feePrice'] = new BigNumber(
+          bufferToHex(tx['gasPrice'])
+        ).toString();
+        newTx['amount'] =
+          new BigNumber(bufferToHex(tx['value'])).toNumber() || 0;
```

Comment:
this has to be toString has well, web3-utils has hexToNumber you can use all across